test/infrastructure/docker/e2e/e2e.conf

```diff
@@ -0,0 +1,56 @@
+---
```

Comment:
I think ideally it would be nice to be able to consume the default config and only provide additional config here, otherwise any time we update the version of cert manager or make other tweaks to the default config they would need to be replicated everywhere the test framework is used.

Reply:
Hi @detiber,

This is a special case because this is using relative paths to load the CAPI, CABPK, and CACPK providers' kustomization roots.

However, the default config **will** work for anyone not running relative to the CAPI provider.

Reply:
Hi @detiber,

FWIW, the whole purpose of the config is to keep things flexible, hence it's not opinionated about what a particular component is. I also used lists and not maps to preserve order of component application. Otherwise, sure, a map key could be used to merge data, but then that just gets messy.

We're trying to provide convenience methods, but there's nothing forcing people to use them. This attempt is generic enough that it can be re-used by **most** providers, but also easily copy-pasta'd by people where one size does **not** fit all.